src/x3f_process.c

```diff
@@ -364,14 +365,19 @@ static void interpolate_bad_pixels(x3f_t *x3f, x3f_area16_t *image, int colors)
 		     bpf23[i], row,
 		     image->columns, image->rows); i++;}
 
-  /* Interpolate over autofocus pixels for sd Quattro.
+  /* Interpolate over autofocus pixels for sd Quattro an sd Quattro H.
      TODO: The positions shouldn't really be hardcoded. */
-  #define SDQ_CAMERAID 40
-  if (x3f_get_camf_unsigned(x3f, "CAMERAID", &cameraid) &&
-      cameraid == SDQ_CAMERAID) {
-    static const grid_t sdq_af_luma = {217, 5641, 16, 1, 464, 3312, 32, 2};
-    static const grid_t sdq_af_chroma = {108, 2820, 8, 1, 232, 1656, 16, 1};
-    const grid_t *g = colors == 1 ? &sdq_af_luma : &sdq_af_chroma;
+  
+  x3f_get_camf_unsigned(x3f, "CAMERAID", &cameraid);
+
```

Comment:
I think it refers it testing with || and using the ? operator subsequently. I would rather set g to NULL first, then test for CAMERAID and colors, and finally test on p and if it's not NULL apply the correction.

Not testing the return value of x3f_get_camf_unsigned(x3f, "CAMERAID", &cameraid) is also a bug. CAMERAID can be absent.